client/gui_rpc_server_ops.cpp

```diff
@@ -908,43 +908,13 @@ static void handle_project_attach(GUI_RPC_CONN& grc) {
         }
     }
 
-	// remove http(s):// at the beginning of project address
-	// there is no reason to connect to secure address project
-	// if we're already connected to the non-secure address
-	// or vice versa
-	// also clear last '/' character if present
-
-    const string http = "http://";
-    const string https = "https://";
-
-    string new_project_url = url;
-    size_t pos = new_project_url.find(http);
-    if (pos != string::npos) {
-        new_project_url.erase(pos, http.length());
-    }
-    else if ((pos = new_project_url.find(https)) != string::npos) {
-        new_project_url.erase(pos, https.length());
-    }
-    if (new_project_url.length() >= 1 && new_project_url[new_project_url.length() - 1] == '/') {
-        new_project_url.erase(new_project_url.length() - 1, 1);
-	}
+    canonicalize_master_url(url);
 
     for (i=0; i<gstate.projects.size(); i++) {
         PROJECT* p = gstate.projects[i];
-        string project_url = p->master_url;
-
-        pos = project_url.find(http);
-        if (pos != string::npos) {
-            project_url.erase(pos, http.length());
-        }
-        else if ((pos = project_url.find(https)) != string::npos) {
-            project_url.erase(pos, https.length());
-        }
-        if (project_url.length() >= 1 && project_url[project_url.length() - 1] == '/') {
-            project_url.erase(project_url.length() - 1, 1);
-        }
+        string project_url = canonicalize_master_url(p->master_url);
```

Comment:
Is it safe to modify `p->master_url` here? Would it have undesired implications outside of the scope of this function?